### ISSUE TITLE
OpenExchangeRate being hit 2x as often as `retrieve_every` configuration.

### ISSUE THREAD

**Issue by dbernheisel** [kipcole9/money#152]
The ExchangeRate Retriever seems to have duplicate timers scheduled whenever `retrieve_every` is configured, this results in the OpenExchangeRates (of whichever configured api module) to fetch rates 2x as often.

I believe it's happening here:
https://github.com/kipcole9/money/blob/7fab9d56fb795897bbcaeb58e12e2c72aabc149d/lib/money/exchange_rates/exchange_rates_retriever.ex#L263-L267

Which both land here:
https://github.com/kipcole9/money/blob/7fab9d56fb795897bbcaeb58e12e2c72aabc149d/lib/money/exchange_rates/exchange_rates_retriever.ex#L329-L333

Let's try to track it, and let's assume `retrieve_every` is 1000ms.

1. 0ms - We initialize the Retriever genserver.
2. 0ms - `schedule_work(0)` casts a message to immediately fetch exchange rates. This is put into mailbox. Expected to process immediately after initializing.
4. 0ms - `schedule_work(1000)` casts another message to fetch exchange rates 1000ms from now.
5. 1ms - Genserver has initialized
6. 2ms - The first `:latest_rates` message is processed in the mailbox, the API is immediately hit and now we `schedule_work(1000)` again, expected around 1002ms (now + 1000ms).
7. 1000ms - The second message in the mailbox is now processed, we hit the API again and schedule another timer for 1000ms, expected at 2000ms (now + 1000ms).
8. 1002ms - The third message in the mailbox is now processed, and we hit the API yet again and schedule another timer for 100ms, expected at 2002ms (now + 1000ms).
9. and so forth. We have 2 duplicate timers at this point past startup.

One solution I've done to avoid this issue is to store a reference of the timer in the genserver state, and whenever scheduling work, you cancel the timer and reset it.

**Comment by kipcole9** [comment]
@dbernheisel thanks again for a fabulously detailed report. I will dig into this later today. Clearly not acceptable to be consuming at twice the expected rate.

**Comment by kipcole9** [comment]
I changed
```elixir
   schedule_work(0) 
   schedule_work(config.retrieve_every)
```
to be simply:
```elixir
   schedule_work(0) 
```
And then let it run for a few iterations and I believe there is now only a single event loop. 

Would you consider trying this by configuring `ex_money` from github?  If you see the correct behaviour then I'll publish a new release right away.

Of course please re-open if the issue isn't fixed as you see it.

**Comment by dbernheisel** [comment]
Yep. I'll report back on Monday

**Comment by kipcole9** [comment]
I've published [ex_money version 5.15.1](https://hex.pm/packages/ex_money/5.15.1) with the following changelog entry:

### Bug Fixes

* Fixes the exchange rate retriever, removing the double retrieval loop.  Thanks to @dbernheisel for the report. Closes #152.

Thanks again for the report!
